x-pack/legacy/plugins/index_management/index.ts

```diff
@@ -4,36 +4,26 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { resolve } from 'path';
-import { Legacy } from 'kibana';
-import { PLUGIN } from './common/constants';
-import { plugin as initServerPlugin, Dependencies } from './server';
-
-export type ServerFacade = Legacy.Server;
-
 export function indexManagement(kibana: any) {
   return new kibana.Plugin({
-    id: PLUGIN.id,
+    id: 'index_management',
     configPrefix: 'xpack.index_management',
-    publicDir: resolve(__dirname, 'public'),
-    require: ['kibana', 'elasticsearch', 'xpack_main'],
+    // TODO: Remove once All dependent apps have migrated to NP
+    config(Joi: any) {
+      return Joi.object({
+        // display menu item
+        ui: Joi.object({
```

Comment:
Is this needed? I don't _think_ `xpack.index_management.ui.enabled` is an existing setting.

Reply:
Actually, I'm not quite sure I follow why the legacy config is needed, since it didn't exist previously. Can you explain?

Reply:
I thought those were necessary but apparently not. I think earlier we forgot to add them though (at least the `enabled` config) 😊 

I removed everything but `configPrefix` and `id` which seems to be the minimum for the dependent apps to not complain.